### PR TITLE
Upgrading to syn 0.12 and quote 0.4

### PR DESCRIPTION
**Pull request by jstnlef**
This is to address https://github.com/pest-parser/pest/issues/203

### CODE REVIEW

**Comment by jstnlef** [comment]
hrm. Looks like there are failing tests.
```
---- tests::derive_ok stdout ----
	thread 'tests::derive_ok' panicked at 'proc_macro::__internal::with_sess() called before set_parse_sess()!', /checkout/src/libproc_macro/lib.rs:822:9
```
I'm really not sure what to make of this. The code runs fine when not run in a testing context but I can't seem to figure out how to get these to work. Any thoughts? Specifically, they fail when trying to create the TokenStream object from the definition string.
```
let tokens = TokenStream::from_str(definition).unwrap();
```

**Comment by dragostis** [comment]
This seems to be a bug in the `proc_macro` crate. I would imagine that `TokenStream::from_str` should not panic.

**Comment by cuviper** [comment]
You could let `parse_derive` and those tests work directly with `DeriveInput` instead:
https://paste.fedoraproject.org/paste/3SaTTY1DfyDCf2UoeP8EGg

**Comment by jstnlef** [comment]
The works for me. I'll make the change tonight.

File: derive/src/lib.rs [hunk 1]
```diff
@@ -332,20 +330,19 @@ fn read_file<P: AsRef<Path>>(path: P) -> io::Result<String> {
...
-            MetaItem::NameValue(ref ident, _) => format!("{}", ident) == "grammar",
+        .filter(|attr| match attr.interpret_meta() {
+            Some(Meta::NameValue(name_value)) => name_value.ident.to_string() == "grammar",
             _ => false
         })
         .collect();
 
     let filename = match grammar.len() {
-        0 => panic!("a grammar file needs to be provided with the #[grammar(\"...\")] attribute"),
+        0 => panic!("a grammar file needs to be provided with the #[grammar = \"...\"] attribute"),
```
Comment (dragostis) [inline derive/src/lib.rs]: Ouch. This is very sad leftover. Glad you found it!

**Pull request merged**